app/src/main/java/org/mozilla/reference/browser/browser/BrowserFragment.kt

```diff
@@ -85,6 +90,8 @@ class BrowserFragment : Fragment(), BackHandler {
         }
 
         promptsFeature = PromptFeature(requireComponents.sessionManager, requireFragmentManager())
+
+        requireComponents.sessionManager.selectedSessionOrThrow.register(this, owner = this)
```

Comment:
There's a trap here. :)

While in the `BrowserActivity` the `BrowserFragment` displays the selected session. However inside `CustomTabActivity` it displays a specific session defined by `sessionId` (See above).

You can see an example of this in the toolbar where we pass down the session id (nullable) and `ToolbarPresenter` uses `SelectionAwareSessionObserver` to display the currently selected session or a fixed session.

Also note that the selected session can change so you may want to use `SelectionAwareSessionObserver` too. At that point it may be worth to wrap that behavior up in a class and not keep it in BrowserFragment directly.

Reply:
Aha, I knew this wasn't the right way it should be done, but I wanted this kind of feedback. Thanks!

Reply:
Maybe a feature that exists only in RB for now?

Reply:
Another reason I put it straight into the fragment was that we needed to pass in a fragment/activity to the feature class in order to call immersive mode, but maybe that can be handled with a callback provided instead.

Reply:
Yeah, just putting this in RB for now seems fine to me. But outside the Fragment would be nice, I guess.

Reply:
I decided to just leg it and put this in `feature-session` instead since it was pretty trivial once I wrote it out as a feature. See: https://github.com/mozilla-mobile/android-components/pull/1617